File: 3799-unique-3-digit-even-numbers/3799-unique-3-digit-even-numbers.py
class Solution:
    def totalNumbers(self, digits: List[int]) -> int:
        from itertools import permutations
        perms = [int(''.join([str(k) for k in list(j)])) for j in permutations(digits,3)]
        count = 0
        counted = set()
        def helper(perms,count,counted):
            if not perms: #some criteria
                return count
            else:
                val = perms.pop(0)
                if val%2==0 and val>=100 and val not in counted:
                    count+=1
                    counted.add(val)
                else:
                    counted.add(val)
                return helper(perms,count,counted)
        
        return helper(perms, count, counted)

File: 3799-unique-3-digit-even-numbers/3799-unique-3-digit-even-numbers.js
/**
 * @param {number[]} digits
 * @return {number}
 */
var totalNumbers = function(digits) {
    function getPermutations(arr, length) {
        if (length==1) return arr.map(d=>[d]);
        const perms = [];
        arr.forEach((val,idx) => {
            const remaining = arr.slice(0,idx).concat(arr.slice(idx+1));
            const innerPerms = getPermutations(remaining, length - 1);
            innerPerms.forEach(p => perms.push([val, ...p]));
        });
        return perms;
    }

    const perms = getPermutations(digits,3).map(p => parseInt(p.join("")));

    const counted = new Set();

    function helper(perms, count, countedSet) {
        if (perms.length===0) return count;
        const val = perms.shift();
        if (val%2===0 && val>=100 && !countedSet.has(val)) {
            countedSet.add(val);
            count+=1;
        } else {
            countedSet.add(val);
        }
        return helper(perms, count, countedSet);
    }
    return helper(perms, 0, counted);
};